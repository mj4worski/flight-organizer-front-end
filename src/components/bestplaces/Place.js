import React from 'react';
import './Place.scss';


const Place = ({ name, shortDescription }) => (
  <div className="place">
    <h2 className="place__name">{name}</h2>
    <p className="place__description">{shortDescription}</p>
  </div>
);

export default Place;
