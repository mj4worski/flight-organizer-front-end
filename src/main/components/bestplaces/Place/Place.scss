.place {
  margin: 30px 0;
  border-radius: 25px;
  height: 330px;
  width: 32%;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  cursor: pointer;
  position: relative;

  &__name {
    text-align: center;
    padding-bottom: 20px;
    font-family: 'Architects Daughter', cursive;
    font-size: 35px;
    color: rgba(52, 45, 111, 0.95);
  }

  &__link {
    text-decoration: none;
  }

}