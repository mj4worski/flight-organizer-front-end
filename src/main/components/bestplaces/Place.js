import React from 'react';
import PropTypes from 'prop-types';
import './Place.scss';

const Place = ({ name, shortDescription }) => (
  <div className="place">
    <h2 className="place__name">{name}</h2>
    <p className="place__description">{shortDescription}</p>
  </div>
);

Place.propTypes = {
  name: PropTypes.string.isRequired,
  shortDescription: PropTypes.string.isRequired,
};

export default Place;
