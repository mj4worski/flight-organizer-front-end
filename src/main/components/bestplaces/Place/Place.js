import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import defaultImage from '../images/default.jpg';
import './Place.scss';

const Place = ({ name, image = defaultImage }) => {
  const sectionStyle = {
    backgroundImage: `url(${image})`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center center',
    backgroundSize: 'cover',
  };
  return (
    <div className="place" style={sectionStyle}>
      <span className="place__front-panel" />
      <Link to="#" className="place__link" >
        <h2 className="place__name">{name}</h2>
      </Link>
    </div>
  );
};

Place.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.any,
};

export default Place;
