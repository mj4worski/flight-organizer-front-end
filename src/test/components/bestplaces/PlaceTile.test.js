import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Place from '../../../main/components/bestplaces/PlaceTile';

it('should render PlaceTile component with default image', () => {
  const component = renderer.create(
    <MemoryRouter >
      <Place name="Warsaw" />
    </MemoryRouter>,
    );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render PlaceTile component with backgroundImage as url-to-image', () => {
  const component = renderer.create(
    <MemoryRouter >
      <Place name="Warsaw" image="url-to-image" />
    </MemoryRouter>,
    );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
