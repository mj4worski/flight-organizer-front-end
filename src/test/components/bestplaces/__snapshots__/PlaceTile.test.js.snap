// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Place component with backgroundImage as url-to-image 1`] = `
<div
  className="place-tile"
  style={
    Object {
      "backgroundImage": "url(url-to-image)",
      "backgroundPosition": "center center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
    }
  }
>
  <span
    className="place-tile__front-panel"
  />
  <a
    className="place-tile__link"
    href="#"
    onClick={[Function]}
  >
    <h2
      className="place-tile__name"
    >
      Warsaw
    </h2>
  </a>
</div>
`;

exports[`should render Place component with default image 1`] = `
<div
  className="place-tile"
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
      "backgroundPosition": "center center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
    }
  }
>
  <span
    className="place-tile__front-panel"
  />
  <a
    className="place-tile__link"
    href="#"
    onClick={[Function]}
  >
    <h2
      className="place-tile__name"
    >
      Warsaw
    </h2>
  </a>
</div>
`;
